update process per SSWG 9/16 meeting (#50)
swift-server/sswg: process/incubation.md

* Communication channels (slack, irc, mailing lists)
* Website (optional)
* Social media accounts (optional)
* Community size and any existing sponsorship (optional)

> Once a proposal PR is submitted, the SSWG will assign a review manager during it's bi-weekely meeting.

Reviewer: ```suggestion
Once a proposal PR is submitted, the SSWG will assign a review manager during it's bi-weekly meeting.
```

### Ecosystem Index

All projects and their respective levels will be listed on the [Swift Server Ecosystem Index](https://swift.org/server/#projects).
In cases where more than one project solves a particular problem (e.g., two similar database drivers), they will be ordered by popularity.
The SSWG reserves the right to define a singular solution for critical building blocks, such as Logging or Metrics APIs, where consistency across the ecosystem is of a critical nature.

It is recommended for projects that have been accepted to any of the maturity levels to list the maturity level in their project's README with the appropriate badge as defined:

[![sswg:sandbox|94x20](https://img.shields.io/badge/sswg-sandbox-lightgrey.svg)](https://github.com/swift-server/sswg/blob/master/process/incubation.md#sandbox-level)
[![sswg:incubating|104x20](https://img.shields.io/badge/sswg-incubating-blue.svg)](https://github.com/swift-server/sswg/blob/master/process/incubation.md#incubating-level)
[![sswg:graduated|104x20](https://img.shields.io/badge/sswg-graduated-green.svg)](https://github.com/swift-server/sswg/blob/master/process/incubation.md#graduated-level)

> The SSWG will meet every 6 month to review all projects, and it reserves the right to demote, archive, or remove projects that no longer fulfill minimal requirements.

Reviewer: ```suggestion
The SSWG will meet every 6 months to review all projects, and it reserves the right to demote, archive, or remove projects that no longer fulfill minimal requirements.
```

* Ecosystem
  * Uses SwiftPM
  * Integrated with critical SSWG ecosystem building blocks, e.g., Logging and Metrics APIs, SwiftNIO for IO
* Longevity
  * Must be from a team that has more than one public repository (or similar indication of experience)
  * SSWG should have access / authorization to graduated repositories in case of emergency
  * Adopt the [SSWG Security Best Practices](#security-best-practices))    
* Testing, CI and Release
  * Have unit tests for Linux
  * CI setup, including testing PRs and the main branch
  * Follow semantic versioning, with at least one published pre-release (e.g. 0.1.0, 1.0.0-beta.1) or release (e.g. 1.0.0)
* Licensing

>   * Apache 2, MIT or BSD (Apache 2 recommended)

Reviewer 2: missing serial comma

>   * Apache 2, MIT or BSD (Apache 2 recommended)

Author: ```suggestion
  * Apache 2, MIT, or BSD (Apache 2 recommended)
```

* Conventions and Style
  * Adopt [Swift API Design Guidelines](https://swift.org/documentation/api-design-guidelines/)

>   * Follow [SSWG Technical Best Practices](#technical-best-practices)) when applicable.

Reviewer 2: extra `)`

>   * Follow [SSWG Technical Best Practices](#technical-best-practices)) when applicable.

Author: ```suggestion
  * Follow [SSWG Technical Best Practices](#technical-best-practices) when applicable.
```

  * Prefer to adopt code formatting tools and integrate them into the CI

## Graduation Requirements

* [Minimal Requirements](#minimal-requirements)
* Have stable API (no pending/planned breaking API changes), with at least one published major release (e.g. 1.0.0)
* Support new GA versions of Swift within 30d
* CI setup for two latest Swift.org recommended versions of Swift
* CI setup for at least one of Swift.org recommended Linux distributions

> * CI setup for each platform supported by the library or tool

Reviewer 3: Swift-NIO runs on anything from Apple Watch up - it doesn't have CI for all the device platforms.  

As the swift server working group are we only interested in potential server platforms eg macOS, linux, windows?

Project authors are also encouraged to make use of their source control system security features (for example: Github's "Security Advisories" and Gitlab's "Confidential Issues") to manage the vulnerabilities and inform their users.

## Technical Best Practices

* Prefer native Swift over C wrapping, where appropriate
* Concurrency / IO
  * Packages should be non-blocking (w/ async API) unless not possible (blocking C libs, etc)
  * There should be as little (preferably no) wrapping of NIO as possible. Exposing NIO types directly will go a long way for making packages compatible.

>   * Blocking code should be wrapped in NonBlockingIOPool (like Vapor's SQLite package)  

Reviewer: ```suggestion
  * Blocking code should be wrapped in [NIOThreadPool](https://apple.github.io/swift-nio/docs/current/NIO/Classes/NIOThreadPool.html) (like Vapor's SQLite package)  
```
I believe this type was renamed in NIO 2.0, switching to new name and adding a link.

> * Uses force unwraps and force tries only as preconditions, ie. conditions that the programmer regards as impossible or programmer error. All force tries/unwraps should come with a comment stating the reasons

Reviewer 3: Use rather than Uses?  Similarly Do rather than Does below?  Feels like the subject of this section changes as you read through it to me although I wouldn't claim expertise at written text.
